t/cobrand/bathnes.t

```diff
@@ -0,0 +1,180 @@
+            'Status',
+            'Latitude',
+            'Longitude',
+            'Query',
+            'Ward',
+            'Easting',
+            'Northing',
+            'Report URL',
+        ],
+        'Column headers look correct';
+};
+
+subtest "Custom CSV fields permission can be granted" => sub {
+    $mech->log_in_ok( $superuser->email );
+
+    is $counciluser->user_body_permissions->count, 0, 'counciluser has no permissions';
+
+    $mech->get_ok("/admin/user_edit/" . $counciluser->id);
+    $mech->content_contains('Extra columns in CSV export');
+
+    $mech->submit_form_ok( { with_fields => {
+        name => $counciluser->name,
+        email => $counciluser->email,
+        body => $counciluser->from_body->id,
+        phone => '',
+        flagged => undef,
+        "permissions[export_extra_columns]" => 'on',
+    } } );
+
+    ok $counciluser->has_body_permission_to("export_extra_columns"), "counciluser has been granted CSV extra fields permission";
+};
+
+subtest 'extra CSV columns are present if permission granted' => sub {
+    $mech->log_in_ok( $counciluser->email );
+
+    $mech->get_ok('/dashboard?export=1');
+
+    open my $data_handle, '<', \$mech->content;
+    my $csv = Text::CSV->new( { binary => 1 } );
+    my @rows;
+    while ( my $row = $csv->getline( $data_handle ) ) {
+        push @rows, $row;
+    }
+    is scalar @rows, 4, '1 (header) + 3 (reports) = 4 lines';
+
+    is scalar @{$rows[0]}, 24, '24 columns present';
+
+    is_deeply $rows[0],
+        [
+            'Report ID',
+            'Title',
+            'Detail',
+            'User Name',
+            'Category',
+            'Created',
+            'Confirmed',
+            'Acknowledged',
+            'Fixed',
+            'Closed',
+            'Status',
+            'Latitude',
+            'Longitude',
+            'Query',
+            'Ward',
+            'Easting',
+            'Northing',
+            'Report URL',
+            'User Email',
+            'User Phone',
+            'Reported As',
+            'Staff User',
+            'Attribute Data',
+            'Site Used',
+        ],
+        'Column headers look correct';
+
+    is $rows[1]->[18], 'normaluser@example.com', 'User email is correct';
+    is $rows[1]->[20], '', 'Reported As is empty if not made on behalf of another user/body';
+    is $rows[1]->[21], '', 'Staff User is empty if not made on behalf of another user';
+    is $rows[1]->[22], 'width = 10cm; depth = 25cm', 'Attribute Data is correct';
+    is $rows[1]->[23], 'iOS', 'Site Used shows whether report made via app';
+
+    is $rows[2]->[18], 'counciluser@example.com', 'User email is correct';
+    is $rows[2]->[20], 'body', 'Reported As is correct if made on behalf of body';
+    is $rows[2]->[21], '', 'Staff User is empty if not made on behalf of another user';
+    is $rows[2]->[22], '', 'Attribute Data is correct';
+    is $rows[2]->[23], 'bathnes', 'Site Used shows correct cobrand';
+
```

Comment:
Is there not also a 'report as anonymous user' option which might as well get tested for completeness.